sdks/python/setup.py

```diff
@@ -359,7 +359,7 @@ def get_portability_package_data():
           'fastavro>=0.23.6,<2',
           'fasteners>=0.3,<1.0',
           # TODO(https://github.com/grpc/grpc/issues/37710): Unpin grpc
-          'grpcio>=1.33.1,<2,!=1.48.0,!=1.59.*,!=1.60.*,!=1.61.*,!=1.62.0,!=1.62.1,<1.66.0',  # pylint: disable=line-too-long
+          'grpcio>=1.65.5,<1.66.0',
```

Comment:
Per https://github.com/apache/beam/pull/32773/files#r1800175195 I think we can at least bump the lower bound down to `1.65.2`, right?

From https://github.com/apache/beam/pull/32784 it looks like we do indeed need this change, but I'd like @tvalentyn to weigh in as well. I don't really love making the boundaries so tight, but it seems like we might not have a choice

Reply:
Based on https://github.com/apache/beam/pull/32773#issuecomment-2415258973, I think the lower bound should be 1.65.5 although it is tight.

Reply:
Ack, I see - thanks

Reply:
> I don't really love making the boundaries so tight

+1. Discussed another suggestion offline on how we could revert the grpcio version bump.